lib/hash_ring/ets.ex

```diff
@@ -92,46 +114,85 @@ defmodule ExHashRing.HashRing.ETS do
 
   @spec get_ring_gen(atom) :: {:ok, integer} | :error
   def get_ring_gen(name) do
-    with {:ok, {_, ring_gen, _}} <- Config.get(name) do
-      {:ok, ring_gen}
+    case Config.get(name) do
+      {:ok, {_, ring_gen, _}} -> {:ok, ring_gen}
+      {:ok, {_, ring_gen, _, _}} -> {:ok, ring_gen}
+      x -> x
     end
   end
 
   @spec find_node(atom, binary | integer) :: {:ok, binary} | {:error, atom}
   def find_node(name, key) do
-    with {:ok, config} <- Config.get(name),
-         {_, node} <- find_next_highest_item(config, Utils.hash(key)) do
-      {:ok, node}
-    else
-      {:error, error} -> {:error, error}
+    case Config.get(name) do
+      {:ok, {table, gen, num_nodes}} when num_nodes > 0 ->
+        find_node_inner(table, gen, num_nodes, key)
+
+      {:ok, {table, gen, num_nodes, overrides}} when num_nodes > 0 ->
+        if override = find_override(overrides, key) do
+          {:ok, override}
+        else
+          find_node_inner(table, gen, num_nodes, key)
+        end
+
+      {:error, error} ->
+        {:error, error}
+
+      _ ->
+        {:error, :invalid_ring}
+    end
+  end
+
+  defp find_node_inner(table, gen, num_nodes, key) do
+    hash = Utils.hash(key)
+
+    case find_next_highest_item(table, gen, num_nodes, hash) do
+      {_, node} -> {:ok, node}
       _ -> {:error, :invalid_ring}
     end
   end
 
   @spec find_nodes(atom, binary | integer, integer) :: {:ok, [binary]} | {:error, atom}
   def find_nodes(name, key, num) do
-    with {:ok, {_, _, num_nodes} = config} when num_nodes > 0 <- Config.get(name),
-         nodes <- do_find_nodes(config, min(num, num_nodes), Utils.hash(key), []) do
-      {:ok, nodes}
-    else
-      {:error, error} -> {:error, error}
-      _ -> {:error, :invalid_ring}
+    hash = Utils.hash(key)
+
+    case Config.get(name) do
+      {:ok, {table, gen, num_nodes}} when num_nodes > 0 ->
+        nodes = do_find_nodes(table, gen, num_nodes, min(num, num_nodes), hash, [])
+
+        {:ok, nodes}
+
+      {:ok, {table, gen, num_nodes, overrides}} when num_nodes > 0 ->
+        nodes = do_find_nodes(table, gen, num_nodes, min(num, num_nodes), hash, [])
+
+        if override = find_override(overrides, key) do
```

Comment:
can’t we just pass the override as a 1 item list to do_find_nodes?

Reply:
ya! not sure why i did this :thinking: